app/src/test/java/it/niedermann/owncloud/notes/shared/util/NoteUtilTest.java

```diff
@@ -99,4 +99,40 @@ public void testGenerateNoteExcerpt_sdk_30() {
         // content has markdown while titles markdown is already stripped
         assertEquals("Title   Bar", NoteUtil.generateNoteExcerpt("# Title\n- Title\n- Bar", "Title"));
     }
+
+    @Test
+    public void testGetContentWithoutCheckedItems() {
+        String newLine = System.getProperty("line.separator");
+
+        String multilineWithCheckboxes = "- [ ] Line A"
+                .concat(newLine)
+                .concat("- [x] Line B")
+                .concat(newLine)
+                .concat("- [X] Line C")
+                .concat(newLine)
+                .concat(newLine)
+                .concat("* [x] Line D")
+                .concat(newLine)
+                .concat("* [ ] Line E")
+                .concat(newLine)
+                .concat("+ [x] Line F")
+                .concat(newLine)
+                .concat("+ [ ] Line G")
+                .concat(newLine)
+                .concat("    + [ ] G1")
+                .concat(newLine)
+                .concat("    + [x] G2");
+
+        String expected = "- [ ] Line A"
+                .concat(newLine)
+                .concat(newLine)
+                .concat("* [ ] Line E")
+                .concat(newLine)
+                .concat("+ [ ] Line G")
+                .concat(newLine)
+                .concat("    + [ ] G1")
+                .concat(newLine);
```

Comment:
As stated above, there are very many cases of the `commonmark` specification not considered (and also not covered by the tests)... @AndyScherzinger  and @tobiasKaminsky will have to decide how compliant is "compliant enough" :wink: